fenicsprecice/fenicsprecice.py

```diff
@@ -273,9 +288,60 @@ def initialize(self, coupling_subdomain, mesh, function_space, write_function=No
             Recommended time step value from preCICE.
         """
 
-        coords = function_space.tabulate_dof_coordinates()
+        write_function_space, write_function = None, None
+        if type(write_object) is Function:  # precice.initialize_data() will be called using this Function
+            write_function_space = write_object.function_space()
+            write_function = write_object
+        elif type(write_object) is FunctionSpace:  # preCICE will use default zero values for initialization.
+            write_function_space = write_object
+            write_function = None
+        elif write_object is None:
+            pass
+        else:
+            raise Exception("Given write object is neither of type dolfin.functions.function.Function or "
+                            "dolfin.functions.functionspace.FunctionSpace")
+
+        if read_function_space is None and write_function_space:
+            self._coupling_type = CouplingMode.UNIDIR_WRITE
+            assert (self._config.get_write_data_name())
+            print("Participant {} is write-only participant".format(self._config.get_participant_name()))
+            mesh = write_function_space.mesh()
+        elif read_function_space and write_function_space is None:
+            self._coupling_type = CouplingMode.UNIDIR_READ
+            assert (self._config.get_read_data_name())
+            print("Participant {} is read-only participant".format(self._config.get_participant_name()))
+            mesh = read_function_space.mesh()
+        elif read_function_space and write_function_space:
+            self._coupling_type = CouplingMode.BIDIR
+            assert (self._config.get_read_data_name() and self._config.get_write_data_name())
+            mesh = read_function_space.mesh()
+        elif read_function_space is None and write_function_space is None:
+            raise Exception("Neither read_function_space nor write_function_space is provided. Please provide a write "
+                            "function_space if one-way coupling with this participant only writing data is intended. "
+                            "Please provide a read_function_space if one-way coupling with this participant only "
+                            "reading data is intended. If two-way coupling is implemented then both read_function_space"
+                            " and write_object need to be provided.")
```

Comment:
```suggestion
            raise Exception("Neither read_function_space nor write_function_space is provided. Please provide a "
                            "write_object if this participant is used in one-way coupling and only writes data. "
                            "Please provide a read_function_space if this participant is used in one-way coupling and "
                            "only reads data. If two-way coupling is implemented then both read_function_space"
                            " and write_object need to be provided.")
```
This exception is triggered correctly. But I realized that what I have suggested before is almost unreadable... Do you think this message is an improvement?